integration_tests/src/main/python/data_gen.py

```diff
@@ -323,8 +323,9 @@ def start(self, rand):
 class FloatGen(DataGen):
     """Generate floats, which some built in corner cases."""
     def __init__(self, nullable=True,
-            no_nans=False, special_cases=None):
+            no_nans=False, special_cases=None, all_nans=False):
```

Comment:
This approach does not scale well to other values we might want to return from a generator.

Rather than add a very specific case to the Float/Double generators, we should leverage the existing `SetValuesGen` which is designed to return values from a hardcoded list of possibilities, e.g.: `SetValuesGen(FloatType(), [math.nan])`.

Reply:
Updated.